main.rb

```diff
@@ -59,31 +56,29 @@ def welcome_screen
   welcome = "*"*welcome.length + "\n" + welcome + "\n" + "*"*welcome.length
 end
 
-def setup_player(num_of_player, name, symbol)
+def setup_player(num_of_player, name)
   print "Please insert the name of the player #{num_of_player}: "
   player_name = gets.strip.capitalize
   player_name = player_name == "" ? name : player_name
-  print "Select your symbol[default: #{symbol}]: "
-  player_sym = gets.strip.upcase
-  player_sym = player_sym == "" ? symbol : player_sym
 
-  return player_name, player_sym
+  return player_name
 end
 
-def make_a_move(player, board)
+def setup_move(player, board)
   begin
     puts "Enter the value from 1 to #{board.size**2}"
     print "#{player.name} your turn: "
     next_move = gets.match(/\d+/)[0].to_i
     raise "Please enter a number in the range." unless (next_move).between?(1, board.size**2)
-    player.play(next_move)
-    raise "Please select an empty position." unless board.fill_table(player.value, player.symbol)
+    raise "Please select an empty position." unless board.read_cell(next_move)
   rescue NoMethodError
     puts "Please enter only numeric values"
     retry
   rescue RuntimeError => e
     puts "#{e}"
     retry
+  else
+    return next_move
   end
 end
```

Comment:
A pretty cool job using `rescue` and handling exceptions 😎

Reply:
nice 👍